api/v3/Payment.php

```diff
@@ -41,6 +41,9 @@
  *   Array of financial transactions which are payments, if error an array with an error id and error message
  */
 function civicrm_api3_payment_get($params) {
+  if (!CRM_Core_Permission::check('access CiviContribute')) {
```

Comment:
Definitely specify it in permissions.php and not here. Then it will correctly check according to the standard api3 rules, which are that check_permissions defaults to FALSE on php, TRUE in ajax/rest.